src/vendorpromo/views.py

```diff
@@ -128,12 +128,34 @@ def create_promo_offer(promo_campaign, products, cost):
     return promo_offer
 
 
+def update_promo_offer(promo_campaign, products, cost):
+    promo_campaign.applies_to.name = promo_campaign.name
+    promo_campaign.applies_to.products.clear()
+
+    for product in products:
+        promo_campaign.applies_to.products.add(product)
+    promo_campaign.applies_to.save()
+
+    update_price = promo_campaign.applies_to.prices.first()
+    update_price.cost = cost
+    update_price.save()
+
+    promo_campaign.save()
```

Comment:
do we need to save the promo_campaign here? it doesn't look like this is updating any fields on there directly

Reply:
If it does not enter the products for loop in won't get saved. So I am just being extra carefull to make sure it is saved at the end of the functions